dramatiq/brokers/redis.py

```diff
@@ -316,12 +320,14 @@ def __next__(self):
                     self.misses = 0
 
                     message = Message.decode(data)
+                    self.queued_message_ids.add(message.message_id)
                     return MessageProxy(message)
                 except IndexError:
                     # If there are fewer messages currently being
                     # processed than we're allowed to prefetch,
                     # prefetch up to that number of messages.
                     messages = []
+                    num_messages = len(self.queued_message_ids)
                     if self.message_refc < self.prefetch:
```

Comment:
Seems like this is a bug since `message_refc` has been removed. Did you mean to use `num_messages` here?